Source/WebCore/workers/service/server/SWServerRegistration.cpp

```diff
@@ -423,4 +424,39 @@ std::optional<ExceptionData> SWServerRegistration::setNavigationPreloadHeaderVal
     return { };
 }
 
+void SWServerRegistration::addCookieChangeSubscriptions(Vector<CookieChangeSubscription>&& subscriptions)
+{
+    for (auto& requestedSubscription : subscriptions) {
+        bool isNewSubscription = true;
+        for (auto& currentSubscription : m_cookieChangeSubscriptions) {
+            if (currentSubscription.name == requestedSubscription.name && currentSubscription.url == requestedSubscription.url) {
+                isNewSubscription = false;
+                break;
+            }
+        }
+
+        if (isNewSubscription)
+            m_cookieChangeSubscriptions.append(WTFMove(requestedSubscription));
```

Comment:
Did we make sure that Vector was the right data structure for this? Seems we need to guarantee uniqueness, which is usually a case for using HashMap/HashSet/ListHashSet.

Reply:
Good point--changed to use HashSet.